components/script/dom/eventsource.rs

```diff
@@ -357,9 +363,14 @@ impl FetchResponseListener for EventSourceContext {
                     return self.fail_the_connection();
                 }
                 self.origin = meta.final_url.origin().ascii_serialization();
+                // Step 15.4 announce the connection and interpret res's body line by line.
                 self.announce_the_connection();
             },
             Err(_) => {
+                // Step 15.2 if res is a network error, then reestablish the connection, unless
+                // the user agent knows that to be futile, in which case the user agent may
+                // fail the connection.
+
                 // The spec advises failing here if reconnecting would be
                 // "futile", with no more specific advice; WPT tests
                 // consider a non-http(s) scheme to be futile.
```

Comment:
nit: this comment is a bit unnecessary now with the spec text right above it, let's just remove the first part of the sentence

```diff
- // The spec advises failing here if reconnecting would be
- // "futile", with no more specific advice; WPT tests
- // consider a non-http(s) scheme to be futile.
+ // WPT tests consider a non-http(s) scheme to be futile.
```